pylearn2/utils/image.py

```diff
@@ -126,6 +126,18 @@ def show(image):
             raise ValueError('image must have either 2 or 3 dimensions but its'
                              ' shape is ' + str(image.shape))
 
+        max_height = 4096
+        max_width = 4096
+
+        if image.shape[0] > max_height:
+            image = image[0:max_height, :, :]
+            warnings.warn("Cropping image to smaller height to avoid crashing "
```

Comment:
I'd just say "Cropping image width to <pixel val>" and leave the rationale for the docstring, but whatever. Indent below is off, though.